build-scripts/generateConfigurationAttributes.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/* tslint:disable */
+
+import * as fs from 'fs'
+// tslint:disable-next-line:no-implicit-dependencies
+import { JSONSchema4 } from 'json-schema'
+import { compile } from 'json-schema-to-typescript'
+import * as packageJson from '../package.json'
+
+const config = [
+    {
+        debugger: 'aws-sam',
+        requestType: 'direct-invoke' as 'direct-invoke',
+        outputFile: 'src/shared/sam/debugger/awsSamDebugConfiguration.gen.ts',
+        imports: ["import * as vscode from 'vscode'"],
+        topLevelClass: 'AwsSamDebuggerConfiguration'
+    }
+]
+
+const header = `
+/* tslint:disable */
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/**
+ * This file was automatically generated by json-schema-to-typescript.
```

Comment:
would prefer to mention `generateConfigurationAttributes.ts` explicitly. When a developer navigates to one of these generated types, it is not obvious where it came from, nor how to influence it. This is magic that one must unwind. Explicitly mentioning the script that generated it, is a helpful shortcut to unwind the magic.

(side note: I'm ambivalent about adding this magic already. We should generate things only if we (1) expect them to change frequently, (2) aren't covered by tests, (3) are voluminous.

Magic is a huge barrier to entry for people trying to understand a codebase.)